config/jobs/kubernetes/sig-node/dra.jinja

```
@@ -100,7 +100,7 @@ presubmits:
         - --repo-root=.
         - --gcp-zone=us-central1-b
         - --parallelism=1
-        - '--label-filter=DRA && Feature: isSubsetOf { OffByDefault, DynamicResourceAllocation } && !FeatureGate:ResourceHealthStatus && !Flaky {%- if not ci %} && !Slow {%- endif %}'
+        - '--label-filter=DRA && Feature: isSubsetOf { OffByDefault, DynamicResourceAllocation } && !Alpha && !Flaky {%- if not ci %} && !Slow {%- endif %}'
```

The `OffByDefault` became wrong here because the feature gates did not enable all off-by-default features at all, only `DynamicResourceAllocation`. This is the root cause of the problem.

Please remove it, then you don't need the `!Alpha`.

---

Also note that we normally put such changes into canary jobs first, try that, then roll out to the actual jobs. Helps prevent mistakes...

As it's already merged into the real jobs let's fix those directly again.

---

Ah, I see. Fixing now.

---

Thanks for the explanation above, it helps clarify the root cause and how the `OffByDefault` label is misleading. 

In the e2e tests I added for ResourceHealthStatus [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/dra_test.go#L823) they have both the `[Feature:DynamicResourceAllocation]` and `[Alpha]` labels. 

My question is: if we only remove `OffByDefault` from the filter, the test would still be selected because it matches `isSubsetOf { DynamicResourceAllocation }`, right? If so, it would likely continue to fail since the `ResourceHealthStatus` feature gate isn't enabled in this job.

It seems like the most direct way to ensure these jobs only run stable, on-by-default tests is to add the `!Alpha` filter to the node jobs.

---

You don't see the `OffByDefault`, it's not getting copied into the test name. But it is set as a Ginkgo label, which is what matters for `--label-filter`. 

@BenTheElder: we may have to extend `--list-tests` because this doesn't show it either - I wasn't sure what the visual markup for labels should be:
```
go test -v ./test/e2e_node -args --list-tests
```

`ginkgo --dry-run` shows them in color.

@Jpsassine: see https://docs.google.com/document/d/1YkU7EusfOZO9jgJbHvCwMCfMZ5CxYMxZHAblpDdKbUQ/edit?tab=t.0#heading=h.kvj5zur6ebww for more information.

---

Actually, didn't you promote to beta with on-by-default? Why were the new tests then failing when they ran in the job?

`DRA && Feature: isSubsetOf { DynamicResourceAllocation } && !Flaky {%- if not ci %} && !Slow {%- endif %}' is still the right fix for the job because it prevents running tests with a dependency on off-by-default features.

---

would be like `go test -v ./test/e2e_node -args --ginkgo.v --ginkgo.silence-skips  --ginkgo.label-filter='Feature: isSubsetOf { OffByDefault, DynamicResourceAllocation } && !Alpha && !Flaky’`

---

A dry run test with the current label filter shows the proper DRA tests are grabbed and only the 1 e2e `ResourceHealthStatus:Disabled` test is covered.

`
go test -v ./test/e2e_node -args --ginkgo.dry-run --ginkgo.v --ginkgo.silence-skips --ginkgo.label-filter='Feature: isSubsetOf { OffByDefault, DynamicResourceAllocation } && !Alpha && !Flaky'
`

It seems like the current approach and label filter in this diff is correct.